modules/021-cni-cilium/hooks/enable_node_routes.go

```diff
@@ -44,14 +46,14 @@ func shouldEnableNodeRoutes(input *go_hook.HookInput) bool {
 	}
 
 	providerRaw, ok := input.Values.GetOk("global.clusterConfiguration.cloud.provider")
-	if !ok {
-		return false
-	}
+	if ok {
```

Comment:
I'd like to see `default: true` here with the ability to override it through the cni configuration secret.
https://github.com/deckhouse/deckhouse/blob/8f7710e5baaabab2624a201e8e9887b69b0b6771/modules/021-cni-cilium/openapi/config-values.yaml#L15

Is there any problem with enabling it by default?

Reply:
We can do this, the cloud routes will be set regardless of the cloud we are in. Until the CNI configuration secret is deployed in the cloud-provider-* module. Which is later in the initialization order than cni-cilium. At which point we'll (perhaps) revert to `enableNodeRoutes: false`. But the routes will already be created in the OS, which may adversely impact networking in the cluster.
How do you envision this?

Reply:
I think cilium will be automatically enabled right after its config is deployed.
Order will not be
```
020-deckhouse -> 021-cni-cilium (with wrong settings) -> 030-cloud-provider-yandex -> 021-cni-cilium (with ok settings)
```
It will be
```
020-deckhouse -> 030-cloud-provider-yandex -> 021-cni-cilium (with ok settings)
```

Reply:
The problem occurs only if cni-cilium is manually enabled in clouds.

Reply:
We've discussed it in Google Meet and agreed that there is no better solution for this issue.